File: frontend/src/api/knowledge.ts
import client from "./client";
import type {
  KnowledgeEntryDetail,
  KnowledgeEntryListResponse,
  KnowledgeEntrySummary,
} from "../types";

export async function listKnowledgeEntries(params: {
  page?: number;
  page_size?: number;
  product_line_code?: string;
  factory_id?: string;
  source_type?: string;
  q?: string;
}): Promise<KnowledgeEntryListResponse> {
  const resp = await client.get("/knowledge/entries", { params });
  return resp.data;
}

export async function getKnowledgeEntry(id: string): Promise<KnowledgeEntryDetail> {
  const resp = await client.get(`/knowledge/entries/${id}`);
  return resp.data;
}

/** Find the CAPA-sourced knowledge entry for a closed 8D, if any. */
export async function findCapaKnowledgeEntry(
  capaId: string,
  opts?: { document_no?: string; product_line_code?: string },
): Promise<KnowledgeEntrySummary | null> {
  const resp = await listKnowledgeEntries({
    source_type: "capa",
    product_line_code: opts?.product_line_code,
    q: opts?.document_no,
    page: 1,
    page_size: 50,
  });
  return resp.items.find((item) => item.source_id === capaId) ?? null;
}

File: backend/app/schemas/knowledge.py
"""Knowledge entry API schemas (US-E2E-01.8)."""
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class KnowledgeEntryListItem(BaseModel):
    entry_id: uuid.UUID
    source_type: str
    source_id: uuid.UUID
    document_no: str
    title: str
    severity: str | None = None
    product_line_code: str
    factory_id: uuid.UUID
    status: str
    embedding_status: str
    embedding_id: uuid.UUID | None = None
    lesson_summary: str | None = None
    tags: list[str] = Field(default_factory=list)
    created_at: datetime

    model_config = {"from_attributes": True}


class KnowledgeEntryDetail(KnowledgeEntryListItem):
    fields: dict[str, Any]
    content_hash: str | None = None
    llm_status: str | None = None
    updated_at: datetime | None = None


class KnowledgeEntryListResponse(BaseModel):
    items: list[KnowledgeEntryListItem]
    total: int
    page: int
    page_size: int
